test/test_inverse.py

```diff
@@ -663,12 +667,179 @@ def test_KFAC_inverse_from_state_dict(
     inv_kfac_new = KFACInverseLinearOperator.from_state_dict(state_dict, kfac)
 
     # check that the two inverse KFACs are equal
     compare_state_dicts(inv_kfac.state_dict(), inv_kfac_new.state_dict())
-    report_nonclose(test_mvp, inv_kfac_new @ test_vec)
+    test_vec = torch.rand(kfac.shape[1])
+    report_nonclose(inv_kfac @ test_vec, inv_kfac_new @ test_vec)
+
+
+"""KFACInverseLinearOperator with EKFACLinearOperator tests."""
+
+
+@mark.parametrize("cache", [True, False], ids=["cached", "uncached"])
+@mark.parametrize(
+    "exclude", [None, "weight", "bias"], ids=["all", "no_weights", "no_biases"]
+)
+@mark.parametrize(
+    "separate_weight_and_bias", [True, False], ids=["separate_bias", "joint_bias"]
+)
+def test_EKFAC_inverse_exactly_damped_matmat(
+    inv_case: Tuple[
+        Module,
+        Union[MSELoss, CrossEntropyLoss],
+        List[Parameter],
+        Iterable[Tuple[torch.Tensor, torch.Tensor]],
+    ],
+    cache: bool,
+    exclude: str,
+    separate_weight_and_bias: bool,
+    delta: float = 1e-2,
+):
+    """Test matrix-matrix multiplication by an inverse (exactly) damped EKFAC approximation."""
+    model_func, loss_func, params, data, batch_size_fn = inv_case
+    dtype = torch.float64  # use double precision for better numerical stability
+    model_func = model_func.to(dtype=dtype)
+    loss_func = loss_func.to(dtype=dtype)
+    params = [p.to(dtype=dtype) for p in params]
+    data = [
+        (
+            (cast_input(x, dtype), y)
+            if isinstance(loss_func, CrossEntropyLoss)
+            else (cast_input(x, dtype), y.to(dtype=dtype))
+        )
+        for x, y in data
+    ]
+
+    if exclude is not None:
+        names = {p.data_ptr(): name for name, p in model_func.named_parameters()}
+        params = [p for p in params if exclude not in names[p.data_ptr()]]
+
+    EKFAC = EKFACLinearOperator(
+        model_func,
+        loss_func,
+        params,
+        data,
+        batch_size_fn=batch_size_fn,
+        separate_weight_and_bias=separate_weight_and_bias,
+        check_deterministic=False,
+    )
+
+    # manual exactly damped inverse
+    inv_EKFAC_naive = (
+        torch.inverse(
+            EKFAC @ torch.eye(EKFAC.shape[0], dtype=dtype, device=EKFAC._device)
+            + delta * torch.eye(EKFAC.shape[0], dtype=dtype, device=EKFAC._device)
+        )
+        .cpu()
+        .numpy()
+    )
+
+    # check that passing a tuple for exact damping will fail
+    with raises(ValueError):
+        inv_EKFAC = KFACInverseLinearOperator(
+            EKFAC, damping=(delta, delta), use_exact_damping=True
+        )
+
+    # use exact damping with KFACInverseLinearOperator
+    inv_EKFAC = KFACInverseLinearOperator(
+        EKFAC, damping=delta, cache=cache, use_exact_damping=True
+    )
+
+    num_vectors = 2
+    X = random.rand(EKFAC.shape[1], num_vectors)
+    # test for equivalence
+    report_nonclose(inv_EKFAC.to_scipy() @ X, inv_EKFAC_naive @ X)
```

Comment:
Could be better to use PyTorch quantities throughout the tests and not mix SciPy with PyTorch.
Now that almost all operators are purely PyTorch, I would like to eliminate as much SciPy contamination as possible. Do you agree?

Reply:
I agree, the only thing that is a bit awkward is that `report_nonclose` only works with numpy arrays, so we always have to convert to numpy first. I just do this for now, but we could maybe consider extending `report_nonclose` to also work with torch tensors.

Reply:
Oh cool, I didn't realise this exists. I used it in this and a few other instances in [this commit](https://github.com/f-dangel/curvlinops/pull/173/commits/4a5230938e77e8eb258b0bbc5321d74fe13d1991).